### ISSUE TITLE
Feature Request: Optional Dark Theme

### ISSUE THREAD

**Issue by Ultrabenosaurus** [AdguardTeam/AdGuardHome#613]
Please can you add an optional dark theme of some sort? There are plenty of examples online to take inspiration from like community-made Atom themes hosted on Github. I have a browser extension that allows me to turn any website "dark" but it basically just inverts CSS colours, so the colour-coded bits in AdGuard Home and the logo in the top-left don't really work. 

<details><summary>Screenshot:</summary>

![image](https://user-images.githubusercontent.com/1901532/53680286-17677b80-3cd1-11e9-9b00-26f3d5bfcafd.png)

</details>

### Your environment

|   Description  |     Value    |
| -------------- | ------------ |
| Version of AdGuard Home server:| 0.93
| How did you setup DNS configuration:| Primary DNS for my Router
| If it's a router or IoT, please write device model:| Raspberry Pi 3 Model B 
| Operating system and version:| latest Raspbian (I updated last night)

### EDIT

For those interested I have published my latest dark theme CSS on userstyles.org for Stylus users: 

https://userstyles.org/styles/177418/adguard-home-dark 

I recommend using the "Customize Settings" button to set the IP address or local domain of your AdGuard Home interface before installing the theme, so that it works straight away. I've also added a setting so you can specify a main font; personally, I quite like [Lato](https://fonts.google.com/specimen/Lato) at the moment. Thanks to @carlbennett for the initial CSS and inspiration to tinker with this.

**Comment by ameshkov** [comment]
Well, the UI is not our top priority at the moment, but we'll take this feature request into account once we start working on the final design.

**Comment by Ultrabenosaurus** [comment]
That's fine, I thought that'd be the case for now :)

**Comment by frenck** [comment]
It would look amazing though...

![image](https://user-images.githubusercontent.com/195327/57011125-5c8c2900-6c00-11e9-83f7-9d31048383c9.png)

**Comment by carlbennett** [comment]
I'm making a userstyle theme for those interested. This is just a work in progress with what I have right now. But for me, this is so much nicer already.

Get the Stylus extension: https://add0n.com/stylus.html

Add this code:
```css
@-moz-document domain("192.168.1.22") {
body { background-color: #222; color: #eee; }
a { color: #aaffaa; }

.header { background-color: #336633; color: #eee; }
.mobile-menu { background-color: inherit; }

.footer { background-color: #113311; }
.footer a:not(.btn) { color: #aaffaa; }

.card { background-color: #333; }
.card-table tr td, .card-table tr th { border-color: #555; }

}
```

Change the IP address to where your AdGuard Home is located at.

:+1:

I'll update this with better code later as I manipulate my custom style code. Here's a preview.

![image](https://user-images.githubusercontent.com/592196/68642812-f0568000-04d4-11ea-9e1c-a599a453e96a.png)

**Comment by ArchiveBase** [comment]
You dont have to tired like that and its opensource as well.
https://chrome.google.com/webstore/detail/dark-reader/eimadpbcbfnmbkopoojfekhnkhdbieeh
https://addons.mozilla.org/en-US/firefox/addon/darkreader/

**Comment by carlbennett** [comment]
I don't want to have to keep flipping a switch. That extension doesn't work well on some pages. I like having finer control over specific pages. I do have that extension already installed and available to me, though.

**Comment by ArchiveBase** [comment]
![image](https://user-images.githubusercontent.com/46082456/68700861-42240680-05b8-11ea-8ccb-8cfdcf516606.png)

**Comment by Ultrabenosaurus** [comment]
@carlbennett I've added a few bits to improve the Query Log page:

```css
body { background-color: #222; color: #eee; }
a { color: #aaffaa; }
.header { background-color: #336633; color: #eee; }
.mobile-menu { background-color: inherit; }
.footer { background-color: #113311; }
.footer a:not(.btn) { color: #aaffaa; }
.card, .rt-tr { background-color: #333; }
.card-table tr td, .card-table tr th { border-color: #555; }
.rt-tr.red { background-color: #5f3333; }
.rt-tr.green { background-color: #335f33; }
.rt-tr.-even { filter: brightness(85%); }
```

I've set specific colours for the red and green rows, which your CSS left as mostly-white, and made the rows alternate with a slightly darker grey. The alternating bit also required specifying a default colour for `.rt-tr` which I tacked onto your `.card` declaration.

![image](https://user-images.githubusercontent.com/1901532/69089188-c9053300-0a3f-11ea-8e2c-1cf12717b334.png)

**Comment by Ultrabenosaurus** [comment]
For those interested I have published my latest dark theme CSS on userstyles.org for Stylus users: 

https://userstyles.org/styles/177418/adguard-home-dark 

I recommend using the "Customize Settings" button to set the IP address or local domain of your AdGuard Home interface before installing the theme, so that it works straight away. I've also added a setting so you can specify a main font; personally, I quite like [Lato](https://fonts.google.com/specimen/Lato) at the moment.

**Comment by carlbennett** [comment]
@Ultrabenosaurus Amazing work! I'll link to that in my original comment.

**Comment by iganeshk** [comment]
Here's my attempt on dark theme for Adguard Home after getting blinded multiple times accessing from phone.

https://github.com/iganeshk/AdGuardHome/tree/dev-personal

![Dark-Mustard](https://raw.githubusercontent.com/iganeshk/AdGuardHome/dev-assets/DarkMustard_Adguard_Home.png)

**Comment by WildByDesign** [comment]
@Ultrabenosaurus Excellent work with this Stylus theme. Thank you for creating and sharing this. I've been using it for the past 3-4 days and it's fantastic.

It appears that a commit in past 24 hours may have messed up your CSS theming specifically regarding the Dashboard menu. Clicking any of the menu items randomly causes that Dashboard menu section to have the background go all black. You can test this on the most recent AGH beta build that was compiled today. These changes will likely make it into stable release in a few days or so, therefore I just wanted to give you a heads up. I really enjoy your theme.

I don't know CSS at all so I cannot help with a suggested fix, but I believe that the following commit may be what caused the changes and requiring some fixes for your Stylus theme. Link: https://github.com/AdguardTeam/AdGuardHome/commit/a5c2ad1b2fa52a071beb759d6b673de73a3d4f4d

**Comment by Ultrabenosaurus** [comment]
@WildByDesign I'm glad you like it, and thanks for the heads up. I only use "stable" releases myself, but I'll keep an eye out for any issues with the CSS in the next few releases and get it fixed ASAP. I had a look through the changes in that commit but couldn't see any HTML or CSS changes.

**Comment by rodalpho** [comment]
Hope this gets added soon, this is 2020, we expect dark mode on all the things!

**Comment by iganeshk** [comment]
> Hope this gets added soon, this is 2020, we expect dark mode on all the things!

give [this](https://github.com/iganeshk/AdGuardHome/) a try meanwhile, building against master and pushing docker images as well

**Comment by WildByDesign** [comment]
@iganeshk Are you able to share some compiled binaries from your sources?
Unfortunately I haven't figured out how to compile from within Windows 10 WSL1 or WSL2. It's just an area that is very unfamiliar to me still.

**Comment by WildByDesign** [comment]
@iganeshk I have been using your builds for a couple of days now and I am absolutely enjoying it. Easy on the eyes, excellent colour combinations, professional looking, etc.

Keep up the great work. Thank you for sharing this with the community. I've updated to updated (v0.100.9-91-g9a81) build from yesterday now as well. It would be much appreciated if you can continue sharing these builds from time to time until AGH has a built-in dark theme.

Hopefully the developers will find a way to merge this into their codebase. Although that would require some sort of drop-down theme selection or slider of some sort to choose colour schemes.

**Comment by iganeshk** [comment]
@WildByDesign glad you're finding it useful. Also just pushed new builds tagging v0.100.9-128-g82aa.

**Comment by planet0** [comment]
+1 for dark mode, at night the sudden bright white hurts - usually try to configure AdGuard Home during the night to avoid any issues when people are trying to use the internet

**Comment by WildByDesign** [comment]
> It appears that a commit in past 24 hours may have messed up your CSS theming specifically regarding the Dashboard menu. Clicking any of the menu items randomly causes that Dashboard menu section to have the background go all black. You can test this on the most recent AGH beta build that was compiled today.

@Ultrabenosaurus  Good news! This issue has resolved itself with today's stable release **v0.101.0**.  

It's strange because every single beta release (all of them) between the past stable release and this current release have had this menu issue with your theme. It must have something to do with the beta channel releases specifically or potentially they landed a last minute menu fix within hours before the stable release, which commits seem may also be the case.

I think the two following commits (last commits prior to stable release) are likely what fixed the weird menu issue:
https://github.com/AdguardTeam/AdGuardHome/commit/7e2ef010399c0a308131c14beca1f35eb9e331fd
https://github.com/AdguardTeam/AdGuardHome/commit/fa07809189a0528469d7102c82c5b9f663566681


Regardless, all is well and your Stylus theme is still flawless and fabulous. Cheers!

**Comment by Ultrabenosaurus** [comment]
@WildByDesign glad to hear it! Hopefully the next time it breaks is due to the AdGuard Home team adding an official dark theme!

**Comment by vordenken** [comment]
Would love to see a darkmode as well, especially since darkreader in firefox doesnt work on the adguard webpage. Another browser addon just for one site seems a bit overkill too.

**Comment by WildByDesign** [comment]
AdGuard Home fans. We need some more **upvotes** to the original comment of this feature request so that the developers can decide based on the amount of upvotes. They quite likely sort the feature requests by the amount of upvotes and determine when and what to implement.

Let's make this happen! :)

**Comment by Ultrabenosaurus** [comment]
For those interested in my dark theme, I am still actively updating it! At least the parts I can see without going through the new installation procedure... 😉 

https://userstyles.org/styles/177418/adguard-home-dark 

I have just published an update for the Tags input system in the New Client modal. I have no idea why the AdGuard team made it this way, but the actual list of possible tags is inserted and deleted from the HTML every time the drop-down menu "opens" and "closes". Even when I right-click on the opened menu, it disappears. If I open the Inspector and then open the menu, clicking in Inspector to expand the HTML closes the menu WHICH DELETES THE HTML!! 

WHY?! :sob: 

I managed to dissect the structure and classes in the end by inserting a breakpoint on all `mousedown` events and then stepping through until the menu appears, then using the element picker tool from Inspector to focus it. That way the menu was locked in time so I could play with it in Inspector! Except when I was finished, for some reason Firefox wouldn't step through or resume the code any more and removing the breakpoint also didn't fix it so I had to reload the page! 

WHY?! :sob: 

Oh well, it's working now. Enjoy!

**Comment by Veldkornet** [comment]
For me, the font is changed everywhere to Times New Roman, or something similar... I’m guessing that’s not the expectation...

**Comment by Ultrabenosaurus** [comment]
> 
> 
> For me, the font is changed everywhere to Times New Roman, or something similar... I’m guessing that’s not the expectation...

@Veldkornet Using my dark theme? I can only imagine it's because I set the default font value on userstyles.org as `inherit` and put the rule at the same point or before AGH actually sets the default font. I'll update the default to `sans-serif` now, but if you click the Customise Settings button on the userstyles.org page before installing the theme you can enter the name of any font you want to use (provided it is installed on your computer). Personally, I really like [Lato](https://fonts.google.com/specimen/Lato).

**Comment by Veldkornet** [comment]
> > For me, the font is changed everywhere to Times New Roman, or something similar... I’m guessing that’s not the expectation...
> 
> @Veldkornet Using my dark theme? I can only imagine it's because I set the default font value on userstyles.org as `inherit` and put the rule at the same point or before AGH actually sets the default font. I'll update the default to `sans-serif` now, but if you click the Customise Settings button on the userstyles.org page before installing the theme you can enter the name of any font you want to use (provided it is installed on your computer). Personally, I really like [Lato](https://fonts.google.com/specimen/Lato).

I changed it to this, now it looks normal:
`font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Helvetica Neue", Arial, sans-serif;`

**Comment by rodalpho** [comment]
It would be great if AGH allowed users to drop a custom CSS file on the filesystem and just use it. I don't want to install another addon just to style this one page.

**Comment by Ultrabenosaurus** [comment]
Apparently the Query Log has blue rows for PTR / "Rewritten by the hosts file rule" entries; now they're dark blue instead of nearly white.

![image](https://user-images.githubusercontent.com/1901532/82742019-bfdb8280-9d50-11ea-8501-a67be4a8d4b7.png)

I don't know how new / common these are supposed to be, but I've never seen them before and no one has complained about my theme missing them, so... at least it's fixed now!

https://userstyles.org/styles/177418/adguard-home-dark

**Comment by KizunaH** [comment]
:point_right: **Optional Dark Theme**
When implement make sure that dark theme compatible with dark reader extensions and browsers built-in dark mode
Eg: chrome://flags > force dark mode for web contents (this has affected with other web apps)

**Comment by michaelb-ae** [comment]
now that github have his dark mode too, can we hope to see one in adguard home too ^^

**Comment by MrDrache333** [comment]
Thought AdGuard already had that...I mean...piHole has that for a Year now?

**Comment by rodalpho** [comment]
No, unfortunately all the posts with dark themes in this github issue are userstyles, not part of AGH.

**Comment by hl2guide** [comment]
Please update AdGuard Home to have a dark mode.

The tool is so useful, please make it easy on our eyes at night.

**Comment by gaetan-p** [comment]
I can only say I'd like a dark mode too :)

**Comment by ameshkov** [comment]
Hey all, we'll definitely do that, but only **after** we finish with the massive redesign #2554

**Comment by gyCfjSnO** [comment]
**Problem Description**
Web page is very bright when viewing at night

*Proposed Solution**
Have a dark mode to reduce brightness

**Comment by rodalpho** [comment]
For anyone using uBlock Origin, I figured out how to (very hackily) invert colors. It looks better, if not perfect. Custom filters:

```
adguard.domain.com##html:style(filter: invert(1))
adguard.domain.com##html:style(background-color: #3e3e42 !important)
```

<img width="1610" alt="ScreenShot 2021-07-06 at 11 33 24 AM" src="https://user-images.githubusercontent.com/747832/124627984-03f2ef00-de4e-11eb-8827-4962de65c2be.png">

**Comment by F3000** [comment]
Any progress here ? 

For desktop abgösse I use "DarkReader" . But for mobile I had no chance until now .

**Comment by Albirew** [comment]
> For anyone using uBlock Origin, I figured out how to (very hackily) invert colors. It looks better, if not perfect. I use uBO in my browser as it's superior to AdGuard Home's limited host-based blocking, and run AGH on my LAN to block ads in mobile apps, streamer boxes, etc.
> 
> Custom filters:
> 
> ```
> adguard.domain.com##html:style(filter: invert(1))
> adguard.domain.com##html:style(background-color: #3e3e42 !important)
> ```

If you want to keep same color set, it's better to use `filter:hue-rotate(180deg) invert(1)`.
Maybe it's me seeing devil in details but this way, colors are closer to intended ones (eg: dark blue becomes light blue instead of light orange)

![Screenshot 2021-09-11 at 22-26-35 AdGuard Home](https://user-images.githubusercontent.com/2805161/132960778-32bb7f30-60d4-4712-9eda-6bc71b1be89b.png)

**Comment by rodalpho** [comment]
Thank you, that does look much better than a straight inversion. Updated my filters!

**Comment by WildByDesign** [comment]
@Albirew @rodalpho These custom filters for uBlock Origin are interesting and look quite nice. It's a good alternative for now until dark mode is officially here.

Would these custom filters work with AdGuard AdBlocker extension as well?
Is the intention to switch `adguard.domain.com` to the device IP address such as `192.168.0.1`?

Thank you both.

**Comment by Albirew** [comment]
> Would these custom filters work with AdGuard AdBlocker extension as well? Is the intention to switch `adguard.domain.com` to the device IP address such as `192.168.0.1`?

Well, I believe it would works since the format is somewhat recognized on every ad blockers.
It's also working with IP, for reference, my filters are like this:
```
192.168.1.13##html:style(filter:hue-rotate(180deg) invert(1))
192.168.1.13##html:style(background-color: #3e3e42 !important)
```

**Comment by WildByDesign** [comment]
@Albirew Thank you. That worked for me and looks quite nice. The only issue is that the top 50% of the page is dark background, the bottom 40% or so is light grey background, and then the footer at the bottom is dark background. 

Do you have a section of light grey at the bottom where the tables for **Top queried domains** and **Top blocked domains** are? Possibly this could be a difference between uBlock Origin and AdGuard Adblocker extensions.

**Comment by Albirew** [comment]
yep, looks like filter doesn't work on background but still work on background inside of window when applying a filter on background... 🤡
i replaced last  rule color with `192.168.1.13##html:style(background-color: #06080c !important)` making the same color everywhere

**Comment by ToNIX-** [comment]
> yep, looks like filter doesn't work on background but still work on background inside of window when applying a filter on background... 🤡 i replaced last rule color with `192.168.1.13##html:style(background-color: #06080c !important)` making the same color everywhere

Thank you for those filters, it looks absolutely amazing!

**Comment by kmanwar89** [comment]
Just installed AdGuard and the first thing I noticed was the eye-bleaching white background. Extremely hard on my eyes, so I'm glad there's an open issue for this, though really surprised it's been **more than three years** to implement something like dark mode. Is UI/UX not a priority, at all?

**Comment by MrDrache333** [comment]
> Just installed AdGuard and the first thing I noticed was the eye-bleaching white background. Extremely hard on my eyes, so I'm glad there's an open issue for this, though really surprised it's been **more than three years** to implement something like dark mode. Is UI/UX not a priority, at all?

For me it was very easy...deinstalled AdGuard and installed piHole. Has everything i need. Including DarkMode.

**Comment by rodalpho** [comment]
If pihole supported DoH natively I would probably switch back too. Shrug, no one solution works for everybody.

**Comment by ToNIX-** [comment]
> For me it was very easy...deinstalled AdGuard and installed piHole. Has everything i need. Including DarkMode.

Adguard is less prone to break IMHO and the update from the web page is so convenient that I ditched Pi-Hole.

**Comment by t3dium** [comment]
> Just installed AdGuard and the first thing I noticed was the eye-bleaching white background. Extremely hard on my eyes, so I'm glad there's an open issue for this, though really surprised it's been **more than three years** to implement something like dark mode. Is UI/UX not a priority, at all?

This is a dns server, I don’t see why ui should be top priority tbf. I’d get dark reader for the time being if you need dark mode, works on all platforms.

**Comment by benisai** [comment]
This issue should be at the top. I review logs to often and the white is killing my eyes. Please give us a dark mode.

**Comment by carlbennett** [comment]
I discovered another GitHub user who also created their own AdGuard dark themes and shared them:

https://docs.theme-park.dev/themes/adguard/
https://github.com/GilbN/theme.park/

**Comment by KaKi87** [comment]
The *Installation* section and the *Setup* page which it links to, don't explain how to actually install the theme on AdGuard Home, though.

**Comment by AlecRust** [comment]
https://docs.theme-park.dev/setup/#stylus-method

**Comment by KaKi87** [comment]
Oh, but that's client-side.

**Comment by AlecRust** [comment]
https://docs.theme-park.dev/setup/#methods

**Comment by BirdInFire** [comment]
> https://docs.theme-park.dev/setup/#methods
 i'm not quite fan to call external code to adguard home, better to download the code review it and add it.
or use the adblock (brave trick) who work well

**Comment by carlbennett** [comment]
A CSS file is pretty benign, it's not like it's JavaScript. The installation methods that @GilbN provides are adequate. I'm using the nginx `sub_filter` method and it works well, but I'm sure the Stylus would be okay too (again, it's pure CSS).

**Comment by BirdInFire** [comment]
> A CSS file is pretty benign, it's not like it's JavaScript. The installation methods that @GilbN provides are adequate. I'm using the nginx `sub_filter` method and it works well, but I'm sure the Stylus would be okay too (again, it's pure CSS).

sure but since the adblock trick do the jobs will not take any risk at all ^^

**Comment by charles-001** [comment]
@rodalpho Your rule changes are not working for me for some reason - has anything changed?

**Comment by BirdInFire** [comment]
> @rodalpho Your rule changes are not working for me for some reason - has anything changed?

No it's work like a charm for me, what's your browser and your adblock ? do you have the right url before the ## ?

**Comment by charles-001** [comment]
> > @rodalpho Your rule changes are not working for me for some reason - has anything changed?
> 
> No it's work like a charm for me, what's your browser and your adblock ? do you have the right url before the ## ?
![Capture](https://user-images.githubusercontent.com/13244625/174014140-55f9b430-9398-48c9-87ed-5079c1c106af.PNG)

I'm using Chrome. I've tried it with uBlock and Adguard... nothing works. I'm using the correct IP for the URL. Isn't this where you put the rules?

**Comment by BirdInFire** [comment]
> > > @rodalpho Your rule changes are not working for me for some reason - has anything changed?
> > 
> > 
> > No it's work like a charm for me, what's your browser and your adblock ? do you have the right url before the ## ?
> > ![Capture](https://user-images.githubusercontent.com/13244625/174014140-55f9b430-9398-48c9-87ed-5079c1c106af.PNG)
> 
> I'm using Chrome. I've tried it with uBlock and Adguard... nothing works. I'm using the correct IP for the URL. Isn't this where you put the rules?

can't help much i do use the brave intergrated adblock (with my change (see past message)) to adapt it.

**Comment by Bugz000** [comment]
> > > > @rodalpho Your rule changes are not working for me for some reason - has anything changed?
> > > 
> > > 
> > > No it's work like a charm for me, what's your browser and your adblock ? do you have the right url before the ## ?
> > > ![Capture](https://user-images.githubusercontent.com/13244625/174014140-55f9b430-9398-48c9-87ed-5079c1c106af.PNG)
> > 
> > 
> > I'm using Chrome. I've tried it with uBlock and Adguard... nothing works. I'm using the correct IP for the URL. Isn't this where you put the rules?
> 
> can't help much i do use the brave intergrated adblock (with my change (see past message)) to adapt it.

i just applied the two lines using brave internal engine and it works great, guys doing something wrong for sure, not sure what though

just dropping by to say, Please for the love of god dark theme... how can you spend hours developing and testing this all the while staring at the vast expanse of eye bleeding white?

the works has even been done for you, just plug it in, there's us who request it - there's an order of magnitude more who would appreciate it but won't say anything - google also proved that displaying dark themes saves energy globally - it should absolutely be common practice to begin with a black panel than white - it should be encouraged on all fronts

i do not need over 90% of the pixels on my 42" monitor to be blaring the highest lumen white light at me for me to read all 500 black pixels of the text that is being displayed... please
Dark mode. it's not that hard.

**Comment by charles-001** [comment]
> did you clicked on the big blue "Apply changes" button? it should be grey like the others when there are no change pending...

I did, yeah. Still not working for me :/

**Comment by charles-001** [comment]
I installed Brave web browser and it works fine. Seems like uBlock/Adguard doesn't work for some reason.

**Comment by Albirew** [comment]
> just dropping by to say, Please for the love of god dark theme... how can you spend hours developing and testing this all the while staring at the vast expanse of eye bleeding white?

when i put my PR for dark theme, it got refused because they already work on a revamp of the web interface with dark theme included:
> Thanks for the contribution, but as you can see that issue is already marked to be resolved in v0.108. We're actually planning a full redesign in that version, see #2554.

**Comment by Bugz000** [comment]
> > just dropping by to say, Please for the love of god dark theme... how can you spend hours developing and testing this all the while staring at the vast expanse of eye bleeding white?
> 
> when i put my PR for dark theme, it got refused because they already work on a revamp of the web interface with dark theme included:
> 
> > Thanks for the contribution, but as you can see that issue is already marked to be resolved in v0.108. We're actually planning a full redesign in that version, see #2554.

yeah, 2 ticked since january 2021...

**Comment by user45876** [comment]
Also just wanted to say hi and 0lease, please add a dark theme. Please.

**Comment by kmanwar89** [comment]
+1 here - can we please get a dark theme?

For those of us OK with bleeding-edge releases, can an alpha or beta be released with the dark theme? We have no information on WHEN the next version will be released, and there's no info in the roadmap.  You might say we're being left in the dark.....(without a dark theme to match!)

**Comment by raykai** [comment]
Seeing as https://adguard-dns.io/ (AdGuard DNS) has a dark now. Can we plz get a dark theme for AdGuard Home? 
It has been requested over 3 years ago.

**Comment by kmanwar89** [comment]
> Seeing as https://adguard-dns.io/ (AdGuard DNS) has a dark theme now. Can we plz get an official dark coption for AdGuard Home? It has been requested over 3 years ago.

Good luck? We should be asking @ainar-g  why this is not a planned feature?

<img width="336" alt="image" src="https://user-images.githubusercontent.com/11323060/187804013-ea0a65a0-c440-44fe-a11f-fff343f7a5f6.png">

Is adding in dark mode *that* contentious?

**Comment by ainar-g** [comment]
@kmanwar89, please don't spread misinformation.  This is a planned feature, and it's being worked on.  Your screenshot is from a duplicate issue.  Duplicate issues are closed as not planned, because they are, well, duplicates.

As for why it's not done yet, the reason is simple and boring.  We have a limited number of developers, and several large projects over which their time is spread.  AdGuard Home's v0.108 redesign, which includes the dark theme, is one of the larger ones.  Throughout 2022 pretty much the entire frontend team (and about ⅔ of the backend team) has been busy with AdGuard DNS, leaving only enough time for small fixes in AdGuard Home.  Now that it's released, both the backend and the frontend teams have more time for AGH, which is something you all might have noticed by us going through the backlog of issues to make sure we don't miss any real issues.

**Comment by kmanwar89** [comment]
> @kmanwar89, please don't spread misinformation. This is a planned feature, and it's being worked on. Your screenshot is from a duplicate issue. Duplicate issues are closed as not planned, because they are, well, duplicates.
> 
> As for why it's not done yet, the reason is simple and boring. We have a limited number of developers, and several large projects over which their time is spread. AdGuard Home's v0.108 redesign, which includes the dark theme, is one of the larger ones. Throughout 2022 pretty much the entire frontend team (and about ⅔ of the backend team) has been busy with AdGuard DNS, leaving only enough time for small fixes in AdGuard Home. Now that it's released, both the backend and the frontend teams have more time for AGH, which is something you all might have noticed by us going through the backlog of issues to make sure we don't miss any real issues.

@BirdInFire  - my intention was *not* to spread "misinformation", simply an honest mistake from a non-developer.  The fact that there are multiple, duplicate issues with hundreds (if not thousands) of people requesting the same feature makes it hard to follow which one is the "correct" issue. It seems this issue (#613) is the most correct one, with other duplicates (4771, 4684) having been linked back to this one. My apologies for screenshotting the wrong issue.

However, that doesn't change the fact that this issue (613) has been open for 3.5 years with 0 traction from the developers other than it's a "planned fix".  I appreciate the effort you all put into it and eagerly await the fix sometime between now and the next 3.5 years.

In the meantime, I've reverted to using Stylus themes to change the colors myself. I'm not much of a web developer anymore, but I wouldn't think modifying CSS/color themes for an application would be *that* much of a drain of developer resources.  Seems there's plenty of people willing to help speed things along - maybe someone submits a PR to get it fixed before the next release?

**Comment by Albirew** [comment]
> maybe someone submits a PR to get it fixed before the next release?

see #4527

**Comment by rodalpho** [comment]
Adguard Home is free open-source software. Demanding features or changes is unlikely to be fruitful.

**Comment by kmanwar89** [comment]
> Adguard Home is free open-source software. Demanding features is unlikely to be fruitful. It doesn't matter how many users want it, unless you're contributing either money or a pull request, the only path forward is to keep this issue alive in a friendly non-demanding way and hope they eventually get around to it. And if they don't, that's OK too. We didn't pay them anything.

I acknowledge the fact that it's free, and appreciate your sentiment, but it seems the developers aren't interested in even accepting valid PR's, such as the #4527 mentioned above.
 
Case in point:

<img width="1047" alt="image" src="https://user-images.githubusercontent.com/11323060/190032860-c607d2ec-20e4-4764-9470-89796753a39a.png">

Even though someone is doing the 3+ year old pending work for them, it doesn't seem like that PR was accepted for obscure, vague reasons only know to the devs.  I would argue that the nature of open source is (or should be) to foster conversation, _constructive_ criticism and accepting valid PR's as they come in. I don't see that happening here - instead it seems contentious to voice my respectful opinions.

I am not a developer and don't have the skills to submit a PR - I can simply give my backing to a valid idea and hope the developers care.  I would purchase the product itself, but lack of dark mode is 100% a non-starter.

**Comment by KaKi87** [comment]
Actually the reply you screenshotted is justified.

Unlike swagger-api/swagger-ui#5964 for example.

**Comment by siewers** [comment]
This
>[...] and accepting valid PRs as they come in.

is not what open source is about. It's a misconception that a PR has to be accepted, just because someone thought it might be a good idea.
I know of open-source projects where maintainers daily have to reject PRs because people think that any idea they might have, is where the project is headed.
There is some weird sense of entitlement where engineers think that, because they put time and effort into something, it's automatically accepted and appreciated. Time and effort do not equal quality or need, which a lot of people tend to believe.

I think the second screenshot is completely on point and politely asks what contributors exactly need to do - discuss it! Otherwise, you end up in a situation where you wasted your own time.

**Comment by TheScenePool** [comment]
My eyes burn every time I open the page on my 1000nit monitor.

**Comment by mattish91** [comment]
> 

I made some minor improvements to the filter for uBlockOrigin to get closer to a better result with less glitches.

(This is basically pitch black)

```
my.adguard-home.url##html:style(filter: invert(1))
my.adguard-home.url##button.btn-success:style(background-color: #a145ff !important; border-color: #a145ff !important)
my.adguard-home.url##span.badge-success:style(background-color: #a145ff !important; color: #000 !important;)
my.adguard-home.url##a.active:style(color: #994a8b !important; border-color: #994a8b !important;)
my.adguard-home.url##div#root:style(background-color: #fff !important;)
```

If there is any interest i might actually clean this up a bit more to keep colors and other things original as filter: invert (1) is a bad choice over actually setting styles correctly. 😂

**Comment by WildByDesign** [comment]
@mattish91 That looks incredible. Thank you for sharing.

**Comment by eduardoaddad** [comment]
https://github.com/AdguardTeam/AdGuardHome/commit/cd76a307900a29db2fdb4088daf4d1c44060bcb8

Dark theme support is coming

**Comment by ainar-g** [comment]
The dark theme has been merged and released to the Edge channel.  We plan to ship it together with the other fixes in v0.107.22.

A thank you to all who followed this long-overdue issue.  Please follow the instructions in #5375 if you have feedback.

**Comment by thecode** [comment]
Looks great:
![image](https://user-images.githubusercontent.com/1858925/213437771-452631b2-8727-49a0-b53c-48cdbd9d3c53.png)

**Comment by castillofrancodamian** [comment]
"Auto" does not indicate that it is a sector to change this mode.
